internal/engine/evaluator.go

```diff
@@ -235,16 +235,30 @@ func (rpe *resourcePolicyEvaluator) Evaluate(ctx context.Context, tctx tracer.Co
 				}
 			}
 
-			// evaluate output expression if the rule was activated
-			if ruleActivated && rule.Output != nil {
+			// TODO: Remove this block when rule.Output field no longer exists
+			//nolint:dupl
+			if rule.Output != nil && ruleActivated { //nolint:staticcheck
+				var output *enginev1.OutputEntry
 				octx := rctx.StartOutput(rule.Name)
-
-				output := &enginev1.OutputEntry{
+				result.Outputs = append(result.Outputs, &enginev1.OutputEntry{
 					Src: namer.RuleFQN(rpe.policy.Meta, p.Scope, rule.Name),
-					Val: evalCtx.evaluateProtobufValueCELExpr(rule.Output.Checked, variables),
+					Val: evalCtx.evaluateProtobufValueCELExpr(rule.Output.Checked, variables), //nolint:staticcheck
+				})
+				octx.ComputedOutput(output)
+			} else if rule.EmitOutput != nil && rule.EmitOutput.When != nil {
+				var output *enginev1.OutputEntry
+				octx := rctx.StartOutput(rule.Name)
+				if ruleActivated && rule.EmitOutput.When.RuleActivated != nil {
+					result.Outputs = append(result.Outputs, &enginev1.OutputEntry{
+						Src: namer.RuleFQN(rpe.policy.Meta, p.Scope, rule.Name),
+						Val: evalCtx.evaluateProtobufValueCELExpr(rule.EmitOutput.When.RuleActivated.Checked, variables),
+					})
+				} else if !ruleActivated && rule.EmitOutput.When.ConditionNotMet != nil {
+					result.Outputs = append(result.Outputs, &enginev1.OutputEntry{
+						Src: namer.RuleFQN(rpe.policy.Meta, p.Scope, rule.Name),
+						Val: evalCtx.evaluateProtobufValueCELExpr(rule.EmitOutput.When.ConditionNotMet.Checked, variables),
+					})
```

Comment:
I don't think you can rely on `ruleActivated` being `false` here. It can be `false` if there are no matching actions as well. Therefore, the `conditionNotMet` output has to be moved up to the loop where we determine that the condition is satisfied or not.

Also, these `if` statements can be simplified a bit for the `RuleActivated` output case. Something like:

```go
if ruleActivated {
  var outputExpr *exprpb.Expr
  switch {
     case rule.Output != nil:
        outputExpr = rule.Output.Checked
     case rule.EmitOutput != nil && rule.EmitOuput.When != nil && rule.EmitOuput.When.RuleActivated != nil:
        outputExpr = rule.EmitOutput.When.RuleActivated.Checked
    }
    
    if outputExpr != nil {
       var output *enginev1.OutputEntry
	   octx := rctx.StartOutput(rule.Name)
	   result.Outputs = append(result.Outputs, &enginev1.OutputEntry{
					Src: namer.RuleFQN(rpe.policy.Meta, p.Scope, rule.Name),
					Val: evalCtx.evaluateProtobufValueCELExpr(outputExpr, variables), //nolint:staticcheck
				})
				octx.ComputedOutput(output)
    }
} 
```